### ISSUE TITLE
CA1508 (Avoid Dead Conditional Code) using statement with previous instantiation gives false positive

### ISSUE THREAD

**Issue by carlos-quintero** [dotnet/roslyn-analyzers#2250]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

2.9.0

#### Diagnostic ID

CA1508 Avoid Dead Conditional Code

#### Repro steps

1. Use VS 2017 15.9.10
2. Create .NET Framework C# project
3. Add NuGet package Microsoft.CodeAnalysis.FxCopAnalyzers 2.9.0
4. Create a custom ruleset with rule CA1508 set to Error instead of None
5. Add this code:

>            OpenFileDialog openFileDialog;
>
>            openFileDialog = new OpenFileDialog();
>
>            using (openFileDialog)
>            {
>             ...
>            }

#### Expected behavior

No error CA1508

#### Actual behavior

Error CA1508
'openFileDialog' is never 'null'. Remove or refactor the condition(s) to avoid dead code

Note: The problem does not happen with other code, but I prefer the above code:

>            OpenFileDialog openFileDialog;
>
>            using (openFileDialog= new OpenFileDialog())
>            {
>             ...
>            }

**Comment by mavasani** [comment]
This analyzer has known false positives and hence is turned off by default. I would recommend you remove the ruleset entry that is turning on this rule. When the false positives are fixed, we will enable this rule by default.

**Comment by neolution-ch** [comment]
Thanks for updating @mavasani. But we noticed, the rule even seems to not exist anymore

![image](https://user-images.githubusercontent.com/5662575/63838892-896a0600-c97e-11e9-974c-6bbb2b478d6c.png)

**Comment by neolution-ch** [comment]
@mavasani yes this worked ;)

**Comment by npuc85** [comment]
This still gives false positives when it's done in 1 line like

```
using (A a = new A())
{
}
```

**Comment by MelbourneDeveloper** [comment]
@carlos-quintero this is a problem in 2.9.7
@mavasani 

![image](https://user-images.githubusercontent.com/16697547/68989278-31b1ab80-0898-11ea-9caf-e226a56916a5.png)

**Comment by abatishchev** [comment]
Same for me in 2.9.7:

Code:
```c#
using (var fabricClient = new FabricClient())
{
}
```
Error:
>'fabricClient = new FabricClient()' is never 'null'

**Comment by abatishchev** [comment]
Any chance this to be fixed?

![image](https://user-images.githubusercontent.com/351644/69371638-17d7e600-0c55-11ea-8b62-721ef03ac9a7.png)

**Comment by neolution-ch** [comment]
@abatishchev You're probably better off reporting it as a own issue and relate to this. It's unlikely that this ticket will be re-opened.

**Comment by lonelydev** [comment]
i saw this happen for the first time ever when i tried opening and building this https://github.com/Microsoft/aspnet-api-versioning solution. I didn't even know this was something that the analyzer did. 
I had to make local changes to get it to build. 
![image](https://user-images.githubusercontent.com/9505396/111068669-a6631d00-84c1-11eb-9ea9-285191b73931.png)
![image](https://user-images.githubusercontent.com/9505396/111068680-b0851b80-84c1-11eb-8600-d09965b981ee.png)
and many more. 
Could anyone tell me how i can get to view the analyzer settings -https://github.com/dotnet/roslyn-analyzers/issues/2250#issuecomment-525647899
i'm using vs2019 community edition.

**Comment by MelbourneDeveloper** [comment]
Yeah, I'm finding that nobody can build my open source repos. They see nothing but errors and give up in disgust. It's cost me a lot of PRs

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by Lokanath-Cell** [comment]
This error is still happening in my code, Most of the using blocks report the problem even when I am using File streams
